src/MongoDB.Driver/LibMongoCrypt/EncryptionSource.cs

```diff
@@ -132,11 +138,63 @@ private IMongoClient CreateMongoCryptDClient(IReadOnlyDictionary<string, object>
             {
                 connectionString = "mongodb://localhost:27020";
             }
-            // todo:
-            // mongocryptdBypassSpawn
-            // mongocryptdSpawnPath
-            // mongocryptdSpawnArgs
+
+            lock (_processSpawnLock)
+            {
+                SpawnMongocryptdIfNecessary(extraOptions);
+            }
             return new MongoClient(connectionString.ToString());
         }
+
+        // todo: case insensitive?
+        private void SpawnMongocryptdIfNecessary(IReadOnlyDictionary<string, object> extraOptions)
+        {
+            if (!extraOptions.TryGetValue("mongocryptdBypassSpawn", out var mongoCryptBypassSpawn) || (!bool.Parse(mongoCryptBypassSpawn.ToString())))
+            {
+                if (!extraOptions.TryGetValue("mongocryptdSpawnPath", out var path))
+                {
+                    //path = string.Empty; // look at the current directory
+                    // todo: where do we need to take this part in case if it's not specified (not in tests)?
+                    path = @"C:\MongoInstances\4.2.0rc3\bin";
+                }
+
+                if (!Path.HasExtension(path.ToString()))
+                {
+                    string fileName = "mongocryptd.exe";
+                    path = Path.Combine(path.ToString(), fileName);
+                }
+
+                try
+                {
+                    using (Process mongoCryptD = new Process())
+                    {
+                        mongoCryptD.StartInfo.UseShellExecute = true;
+                        mongoCryptD.StartInfo.FileName = path.ToString();
+                        mongoCryptD.StartInfo.CreateNoWindow = false;
+                        if (extraOptions.TryGetValue("mongocryptdSpawnArgs", out var mongocryptdSpawnArgs))
+                        {
+                            // todo: refactor. Consider a enumerable object in `mongocryptdSpawnArgs`.
```

Comment:
@jyemin, what object a user can add into `extraOptions` for `mongocryptdSpawnArgs`?

Currently, I expect a `string`, but probably a user can provide `Dictionary<string, ???>` or `List<string>`?  Should we expect it?

Reply:
done